### PR TITLE
ntru1229

### PR DESCRIPTION
**Pull request by baentsch**
Follows https://github.com/open-quantum-safe/liboqs/pull/1126. CI will only succeed after that has merged.

##### Checklist
- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by baentsch** [comment]
Interop test will pass when https://github.com/open-quantum-safe/boringssl/pull/79 has been merged

File: include/openssl/obj_mac.h [hunk 1]
```diff
@@ -5203,511 +5203,527 @@
 
 #define SN_p256_frodo640aes             "p256_frodo640aes"
 #define LN_p256_frodo640aes             "p256_frodo640aes"
-#define NID_p256_frodo640aes            1237
+#define NID_p256_frodo640aes            1239
```
Comment (dstebila) [inline include/openssl/obj_mac.h:5206]: Are these NID values part of the external API or only ever used internally?  If part of the external API, then renumbering would mean different behaviour for external users of the library if they don't recompile.

**Review by dstebila — APPROVED** [review]

File: include/openssl/obj_mac.h (hunk 1 above)
Comment (baentsch) [reply include/openssl/obj_mac.h:5206]: AFAIK NIDs are OSSL internal references to algorithms that are efficient in lookups (unlike say OIDs) and --again, for all I know-- should never be used/become visible to outside observers (unlike OIDs or group identifiers/"code points"). Thus I consider these changes invisible to "the unwary" not causing interop issues (as regularly tested) nor recompilation requirements when changed. But of course I may be wrong/misunderstand things.
But then again, these NIDs are assigned by a "stock-OpenSSL" script (`make generate_crypto_objects`). And given the many mechanisms within OSSL to keep (other) things stable, I take the way with which this script assignes NIDs as "blessed by the gods of OpenSSL" -- and thus wouldn't question --let alone change-- those assignments :-)

Comment (dstebila) [reply include/openssl/obj_mac.h:5206]: That's what I thought, but just wanted to check I didn't misunderstand. Resolved.

**Review by dstebila — APPROVED** [review]

**Pull request merged**